.eslintrc.js

```diff
@@ -0,0 +1,28 @@
+module.exports = {
+  root: true,
+  env: {
+    node: true,
+  },
+  parserOptions: {
+    ecmaVersion: 2021,
+  },
+  plugins: ['only-warn'],
+  extends: ['eslint:recommended', 'plugin:prettier/recommended', 'plugin:storybook/recommended'],
+  rules: {
+    // Set line length to 120
+    'max-len': ['error', { code: 120 }],
```

Comment:
If prettier is controlling the line-length, why does eslint also need to set the same value?  Shouldn't formatting be left entirely up to prettier?

Reply:
It's simple, we reduce count of vertical lines in IDEs that is created by different plugins (EditorConfig, ESLint, Prettier) and set it all to same value.

## Example

### Good

Single line:

![image](https://user-images.githubusercontent.com/2821574/148294263-6e4c6d5a-9a3c-44a7-838d-0bff70f8a761.png)

### Bad

Multiple lines:

![image](https://user-images.githubusercontent.com/2821574/148294336-53c86dc7-7447-41a0-8da9-68cb4bf86392.png)

Reply:
Sorry, I don't understand, those two screenshots look like completely different import statements.

By using `plugin:prettier/recommended`, you've disabled all of eslint's style rules, so that prettier can handle them.  But then you turn back on the max-len rule, which is going to fight with prettier because they might break lines up differently.  Better to keep all style formatting within prettier, and let eslint only warn about syntax and best-practices.

Reply:
@IanVS 

> those two screenshots look like completely different import statements, and I don't see how they relate to eslint or prettier.

First screenshot from `storybook-builder-vite` (this pull request), second from `storybook` (`next` branch). 

Plugins of a JetBrains IDE automatically getting configs from ESLint and Prettier.

So, in the Storybook we have two different values of max line lenght, one from the Prettier, and another from imported nested config of the ESLint.

But in this pull request I set the same value for all plugins (EditorConfig, ESLint and Prettier), so you see the one vertical line.

> By using plugin:prettier/recommended, you've disabled all of eslint's style rules, so that prettier can handle them.

Yes, I do.

> But then you turn back on the max-len rule, which is going to fight with prettier because they might break lines up differently.

I don't think so, because I set the same value — `120`.

> Better to keep all style formatting within prettier, and let eslint only warn about syntax and best-practices.

Impossible, but I done it in this pull request.

Right now all done as you described:

* we keeping all style formatting within Prettier plugin;
* and warn  about syntax and best practices by using other plugins.

All in one command:

```
eslint
```